file.go

```diff
@@ -10,3 +23,68 @@ type File struct {
 	Filename    string `json:"filename"`
 	Size        int    `json:"size"`
 }
+
+var quoteEscaper = strings.NewReplacer("\\", "\\\\", `"`, "\\\"")
+
+func escapeQuotes(s string) string {
+	return quoteEscaper.Replace(s)
+}
+
+// UploadFile uploads a file to be used as an attachment.
+// See: https://docs.nylas.com/reference#upload
+func (c *Client) UploadFile(
+	ctx context.Context, filename string, file io.Reader,
+) (File, error) {
+	g, ctx := errgroup.WithContext(ctx)
+	req, err := c.newUserRequest(ctx, http.MethodPost, "/files", nil)
+	if err != nil {
+		return File{}, err
+	}
+
+	r, w := io.Pipe()
+	m := multipart.NewWriter(w)
+	req.Body = r
+	req.Header.Set("Content-Type", m.FormDataContentType())
+
+	g.Go(func() (err error) {
+		defer w.Close() // nolint: errcheck
+		defer m.Close() // nolint: errcheck
+
+		sniff := make([]byte, 512)
+		if _, err := file.Read(sniff); err != nil {
+			return err
+		}
+		contentType := http.DetectContentType(sniff)
+
+		h := make(textproto.MIMEHeader)
+		h.Set("Content-Disposition",
+			fmt.Sprintf(`form-data; name="file"; filename="%s"`,
+				escapeQuotes(filename)))
+		h.Set("Content-Type", contentType)
```

Comment:
Yeah, we can't set the Content-Type header if we use that helper :(

I'm actually going to change this to use the file extension instead to avoid the content sniffing